homeassistant/components/axis/config_flow.py

```diff
@@ -99,21 +103,23 @@ async def async_step_user(self, user_input=None):
             except CannotConnect:
                 errors["base"] = "cannot_connect"
 
-        data = self.discovery_schema or {
-            vol.Required(CONF_HOST): str,
-            vol.Required(CONF_USERNAME): str,
-            vol.Required(CONF_PASSWORD): str,
-            vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
-        }
+        data_schema = self.discovery_schema or vol.Schema(
```

Comment:
I'm confused by this change? Why it is wrapped in a schema now? Was it broken before?

Reply:
It wasn't broken before, but `self.discovery_schema` was untyped and it seemed before to type it directly as a Schema than as a dict.

Previously:
- three ways to create a dict, which was then used to create a schema in show_form `data_schema=vol.Schema(data)`

Now:
- three ways to create a schema, which is passed to show_form `data_schema=data_schema`

Reply:
`vol.Schema` can take pretty much anything in it's initialisation.

I guess I could revert this change, and add a type hint like this `discovery_schema: dict[vol.Required, Type] | None`  but I am not convinced it makes it more readable.

Is there a cost involved to creating the schema directly?

Reply:
> Is there a cost involved to creating the schema directly?

It's unneeded wrapping. My point is: If this is needed, the typing upstream might be incorrect and should be fixed first.